go/cmd/ocitool/pull_cmd.go

```diff
@@ -6,18 +6,20 @@ import (
 	"github.com/DataDog/rules_oci/go/pkg/ociutil"
 
 	"github.com/containerd/containerd/images"
+	"github.com/containerd/log"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
-	log "github.com/sirupsen/logrus"
 	"github.com/urfave/cli/v2"
 	orascontent "oras.land/oras-go/pkg/content"
 )
 
 func PullCmd(c *cli.Context) error {
-	resolver := ociutil.DefaultResolver()
-
 	ref := c.Args().First()
 
-	name, desc, err := resolver.Resolve(c.Context, ref)
+	ctx := log.WithLogger(c.Context, log.G(c.Context).WithField("pull-ref", ref))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** I'm slightly weary of this change to use `containerd/log` when it's mentioned that [the package is not intended for use outside of `containerd`](https://github.com/containerd/log/blob/0fc1e28871fdf2786e2cc51bbe4133db6547a199/README.md?plain=1#L5). Given our experience with ignoring these sorts of warnings, it might be worth trying to find another way to go here. I assume it's for consistency in the log output between what we do in `rules_oci` and what they do in `containerd`? Maybe staving off this change until there's movement on https://github.com/containerd/log/issues/2?

In any case, not worth blocking. But something to think about, and do whatever you think is best.

Reply:
That's a fair comment; I think the fact that containerd/log is using logrus _leaks_ because we want to set the log level, and to do that we _have_ to know that containerd is using logrus, at which point we may as well use containerd/log's extra sugar that smuggles fields in the Context.  This is not a _great_ justification: sorry.

Reply:
Nah, no worries. Again, not something to block on. This repo is small enough (and I think contained enough) that if something does break because of `containerd` changing this stuff, we ought to be able to address it without expending a tremendous amount of effort.

Follow your heart!